src/machine/machinectl.c

```diff
@@ -117,12 +118,63 @@ static int compare_machine_info(const void *a, const void *b) {
         return strcmp(x->name, y->name);
 }
 
+static void clean_machine_info(MachineInfo *machines, size_t n_machines) {
+        size_t i;
+
+        if (!machines || !n_machines)
```

Comment:
we generally try to avoid comparing numeric variables with 0 by using C's degrade-to-bool feature, see CODING_STYLE. It's OK to use the degrade-to-bool feaure for ptrs and for ints that are used as bools. Long story short, please change the above to:

```c
if (!machines || n_machines == 0)
```

or something like that.